### PR TITLE
feat: use doc strings from the `abi` in `abigen`

### PR DESCRIPTION
**Pull request by hal3e**
closes: https://github.com/FuelLabs/fuels-rs/issues/1335

Adds ABI doc strings to contract methods and script's main function.

### Checklist
- [x] I have linked to any relevant issues.
- [x] I have added tests that prove my fix is effective or that my feature works.
- [x] I have added necessary labels.
- [x] I have done my best to ensure that my PR adheres to [the Fuel Labs Code Review Standards](https://github.com/FuelLabs/rfcs/blob/master/text/code-standards/external-contributors.md).
- [x] I have requested a review from the relevant team or maintainers.

### CODE REVIEW

File: Cargo.toml
Comment (segfault-magnet) [inline Cargo.toml:108]: I'd make a release before merging this.

Comment (segfault-magnet) [inline Cargo.toml:108]: I'd make a release before merging this.

**Review by segfault-magnet — COMMENTED** [review]
Cool.

Fine with the impl, just waiting for the release of fuel abi types.

**Review by segfault-magnet — APPROVED** [review]

**Review by Salka1988 — APPROVED** [review]

**Pull request merged**
